File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        if len(s) == 1:
            return False

        tempStack = []

        for i in range(len(s)):
            if s[i] == "(" or s[i] == "[" or s[i] == "{":
                tempStack.append(s[i])
            elif s[i] == ")":
                if len(tempStack) != 0 and tempStack[len(tempStack)-1] == "(" :
                    tempStack.pop()
                else:
                    tempStack.append(")")
            elif s[i] == "}":
                if len(tempStack) != 0 and tempStack[len(tempStack)-1] == "{":
                    tempStack.pop()
                else:
                    tempStack.append("}")
            elif s[i] == "]":
                if len(tempStack) != 0 and tempStack[len(tempStack)-1] == "[":
                    tempStack.pop()
                else:
                    tempStack.append("]")
        
        if len(tempStack) == 0:
            return True
        else:
            return False

        

        

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
static const bool Booster = [](){
    #pragma GCC optimize("OFast")
    std::ios_base::sync_with_stdio(0);
    std::cin.tie(nullptr);
    std::cout.tie(nullptr);
    return true;
}();


class Solution {
public:
    bool isValid(string s) {

        bool answer = false;

        vector <char> myVector;

        for(int i=0;i<s.size();i++){
            if(s[i] == '(' || s[i] == '{' || s[i] == '['){
                myVector.push_back(s[i]);
            }

            if(s[i] == ')'){
                if(myVector.size()!= 0 && myVector[myVector.size()-1] == '('){
                    myVector.pop_back();
                }else{
                    myVector.push_back(s[i]);
                }
            }

            if(s[i] == '}'){
                if(myVector.size()!= 0 && myVector[myVector.size()-1] == '{'){
                    myVector.pop_back();
                }else{
                    myVector.push_back(s[i]);
                }
            }

            if(s[i] == ']'){
                if(myVector.size()!= 0 && myVector[myVector.size()-1] == '['){
                    myVector.pop_back();
                }else{
                    myVector.push_back(s[i]);
                }
            }
        }

        if(myVector.size() == 0){
            answer = true;
        }

        return answer;

    }
};